### ISSUE TITLE
Why am i getting IlligalStateException in some andoird devices

### ISSUE THREAD

**Issue by prime-zs** [androidx/media#483]
here is the exception

Exception java.lang.IllegalStateException:
  at androidx.media3.common.util.Assertions.checkState (Assertions.java:100)
  at androidx.media3.session.MediaController.verifyApplicationThread (MediaController.java)
  at androidx.media3.session.MediaController.removeListener (MediaController.java:1786)
  at com.prime.media.core.playback.RemoteImpl$events$1$1.invoke (Remote.kt:160)
  at com.prime.media.core.playback.RemoteImpl$events$1$1.invoke (Remote.kt:159)
  at kotlinx.coroutines.channels.ProduceKt.awaitClose (Produce.kt:54)
  at kotlinx.coroutines.channels.ProduceKt$awaitClose$1.invokeSuspend (Produce.kt)
  at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith (ContinuationImpl.kt:33)
  at kotlinx.coroutines.DispatchedTask.run (DispatchedTask.kt:104)
  at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely (CoroutineScheduler.java:584)
  at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask (CoroutineScheduler.kt:793)
  at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker (CoroutineScheduler.kt:697)
  at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run (CoroutineScheduler.kt:684)

and here is the code

 @OptIn(DelicateCoroutinesApi::class)
    override val events: Flow<Player.Events?> = callbackFlow {
        val observer = object : Player.Listener {
            override fun onEvents(player: Player, events: Player.Events) {
                trySend(events)
            }
        }
        // init browser
        val browser = fBrowser.await()
        // init first one.
        trySend(null)
        // register
        browser.addListener(observer)
        // un-register on cancel
        awaitClose {
            browser.removeListener(observer)
        }
    }.shareIn(
            // what show I use to replace this.
            GlobalScope,
            // un-register when subscriber count is zero.
            SharingStarted.WhileSubscribed(2000, replayExpirationMillis = 5000),
            //
            1
        )

**Comment by tianyif** [comment]
Hi @prime-zs,

I'm not very familiar with kotlinx coroutines, but could you please make sure that the `browser.removeListener(Player.Listener)` is called on the application thread? Per the [documentation](https://developer.android.com/reference/androidx/media3/session/MediaController#threading-model):

```
Methods of this class should be called from the application thread associated with the application looper. Otherwise, `IllegalStateException` will be thrown.
```
Exceptionally, for `browser.addListener(Player.Listener)`, we allow calls to this method from any thread. But for `browser.removeListener(Player.Listener)`, we do [enforce](https://github.com/androidx/media/blob/release/libraries/session/src/main/java/androidx/media3/session/MediaController.java#L1786) that.

Thanks!

**Comment by prime-zs** [comment]
Thanks for replying quickly. I tried to add flowOn(Dispatchers.Main). It seems to work for now on my test device. By the way, the earlier 1.0.0 beta versions were working without any issues. However, after updating to 1.0.0 stable and above, the error first appeared on test devices in the Google Play console.

**Comment by tianyif** [comment]
Yes, we added `verifyApplicationThread()` check in `removeListener` from 1.0.0-rc01, the version (1.0.0-beta03) [doesn't have this check](https://github.com/androidx/media/blob/1.0.0-beta03/libraries/session/src/main/java/androidx/media3/session/MediaController.java#L1752).
